docs/spec/nmt-wrapper.md

```diff
@@ -0,0 +1,84 @@
+In this specification, we elaborate on the design and structure of the Namespace Merkle Tree wrapper.
+
+
+
+## Namespaced Merkle Tree Wrapper
+Namespaced Merkle Tree Wrapper is used to represent a row or column of Celestia extended data square.
+At its core, it is an [NMT](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md) and is defined by a similar set of parameters.
+Namely, the [namespace ID size `NamespaceIDSize`](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md#nmt-data-structure), 
+the underlying hash function for the digest calculation of the [namespaced hashes](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md#namespaced-hash), 
+and the [`IgnoreMaxNamespace` flag](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md#ignore-max-namespace) which dictates how namespace range of non-leaf nodes are calculated.
+In addition, the NMT wrapper is configured with the data square size `SqaureSize` (`k` in the above example), and the index of the row or column it represents `AxisIndex` which is a value in `[0, SquareSize)`.
+These additional configurations are used to determine the namespace ID of the shares that the NMT wrapper represents based on the quadrants to which they belong.
+
+NMT wrapper supports [Merkle inclusion proof](#link-to-the-nmt-spec-for-the-inclusion-proof) for the given share index and [Merkle range proof](#link-to-the-nmt-spec-for-the-range-proof) for a range of share indices.
+It extends the NMT data insertion behaviour (i.e., the [`Push` method](https://github.com/celestiaorg/nmt/blob/master/spec/nmt.md#add-leaves)) to prepend shares with proper namespace before inclusion in the tree.
+
+### Namespace ID assignment to the shares of an extended data square
+To understand the NMT wrapper data structure, it's important to first understand how shares are namespaced in Celestia.
+The namespace ID of a share depends on the quadrant it belongs to.
+
+Shares in the original data square `Q0` already carry their namespace IDs, which are located in the first `NamespaceIDSize` bytes of the share. 
+Thus, the namespace ID of a share in `Q0` can be extracted from the first `NamespaceIDSize` bytes of the share.
+
+However, shares in `Q1`, `Q2`, and `Q3` (that are erasure-coded versions of `Q0` and known as parity shares) do not have any namespace IDs by default. 
+These shares must be assigned a reserved namespace ID, which is called `ParitySharesNamespaceID`.
+`ParitySharesNamespaceID` corresponds to the maximum value representable by `NamespaceIDSize` bytes.
+
+In Celestia, as of the writing of this specification, the `NamespaceIDSize` (which maps to the [`NAMESPACE_ID_BYTES`](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/consensus.md#constants) constant) is `8` bytes, and therefore, [`ParitySharesNamespaceID`](https://github.com/celestiaorg/celestia-app/blob/specs-staging/specs/src/specs/consensus.md#reserved-namespace-ids)  is represented by `8` bytes of `0xFF`.
```

Comment:
[non-blocking]

Instead of specifying 8 here, perhaps we could simply say that the parity namespace is the lexicographically last namespace. Either that, or update to 33 now cc @rootulp

Reply:
I have rephrased this part to eliminate the use of a specific number for the namespace ID size in Celestia. By doing this, we can ensure that this specification does not need to be updated every time there is a change to the namespace size in Celestia. Please see https://github.com/celestiaorg/celestia-app/pull/1442/commits/d0519e7661a14c510b47de30a361ca7a23c55075.